definition/IOmnichannelVisitor.ts

```diff
@@ -0,0 +1,104 @@
+
+export enum ChannelType {
+	LIVECHAT = 'livechat',
+	EMAIL = 'email',
+	SMS = 'sms',
+	WHATSAPP = 'whatsapp',
+	TWITTER = 'twitter',
+	FACEBOOK_MESSENGER = 'facebook-messenger',
+	TELEGRAM = 'telegram',
+	INSTAGRAM = 'instagram',
+	APPLE = 'apple'
+}
+
+export interface IOmnichannelVisitor {
+	_id: string;
+	username: string;
+	token: string;
+
+	name?: string;
+
+	channels?: Array<IIncomingChannel>;
+
+	ts: Date;
+	_updatedAt: Date;
+}
+
+export interface IIncomingChannel {
+	type: ChannelType;
+}
```

Comment:
Hi @KevLehman Yeah, I really liked this idea -  it's making the interfaces abstract and still allowing apps to store all the info it needs to store.  I just had a few comments.

1. Can we keep the name property optional?, because not all channels might provide this info. Eg: Sometimes, Instagram doesn't provide us the name of the user when they have their account private 
2. Trying to combine the discussion we had on DM, where we'll allow rendering of custom data on Room Info panel
```
export interface IIncomingChannel {
	source: ChannelType;
	sourceId: string // this should be the ID used for communication (phone number, userId, etc)
	name?: string; // username or name of user
	otherData: {
	      roomHeaderFields: Array<{ label: string; value: string }>  // this data will be displayed in the room-info panel of a livechat room. (eg: please refer to the sample example image below)
	}
}
```

Example of **roomHeaderFields** in actions
![image](https://user-images.githubusercontent.com/34130764/131292445-a8ef747e-5da0-4f87-ad60-fc2e2e3f171f.png)